### PR TITLE
MGMT-9357 update assisted-service golang version to 1.17

### PR DESCRIPTION
**Pull request by filanov**

### CODE REVIEW

**Comment by filanov** [comment]
skipped ocm 2.4 and 2.3 is that OK?

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml [hunk 1]
```diff
@@ -23,22 +23,22 @@ base_images:
     name: test
     namespace: ocp-kni
     tag: dev-scripts
-  openshift_release_golang-1.16:
+  openshift_release_golang-1.17:
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-cloud_hotfix_releases.yaml:26]: ``cloud_hotfix_releases`` is still pointing to ``2.1.x``, so maybe we want to remain with the same golang version in this file

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by osherdp** [comment]
/hold
(waiting for optional rehearse jobs)

**Review by osherdp — APPROVED** [review]

**Comment by osherdp** [comment]
@filanov results look okay in my opinion
unhold whenever you want

**Pull request merged**